src/components/DepositWidget/useRowActions.tsx

```diff
@@ -16,13 +17,71 @@ interface Params {
 
 interface Result {
   enableToken: (tokenAddress: string) => Promise<void>
-  deposit: (amount: BN, tokenAddress: string) => Promise<void>
-  requestWithdraw: (amount: BN, tokenAddress: string) => Promise<void>
-  claim: (tokenAddress: string) => Promise<void>
+  depositToken: (amount: BN, tokenAddress: string) => Promise<void>
+  requestWithdrawToken: (amount: BN, tokenAddress: string) => Promise<void>
+  claimToken: (tokenAddress: string) => Promise<void>
+  // State
+  enabling: TokenLocalState['enabling']
+  highlighted: TokenLocalState['highlighted']
+  claiming: TokenLocalState['claiming']
+}
+
+interface TokenLocalState {
+  enabling: Map<string, string>
+  highlighted: Map<string, string>
+  claiming: Map<string, string>
+}
+
+const SET_ENABLING = 'enabling'
+const SET_CLAIMING = 'claiming'
+const SET_HIGHLIGHTED = 'highlighted'
+const SET_HIGHLIGHTED_AND_CLAIMING = 'highlighted_and_claiming'
+
+type Actions =
+  | { type: typeof SET_ENABLING; payload: string }
+  | { type: typeof SET_CLAIMING; payload: string }
+  | { type: typeof SET_HIGHLIGHTED; payload: string }
+  | { type: typeof SET_HIGHLIGHTED_AND_CLAIMING; payload: string }
+
+const initialState: TokenLocalState = {
+  enabling: new Map(),
+  highlighted: new Map(),
+  claiming: new Map(),
+}
+
+const initState = (): TokenLocalState => initialState
+
+const reducer = (state: TokenLocalState, action: Actions): TokenLocalState => {
+  switch (action.type) {
+    case SET_ENABLING:
+    case SET_CLAIMING:
+    case SET_HIGHLIGHTED:
+      return {
+        ...state,
+        [action.type]: state[action.type].has(action.payload)
+          ? state[action.type].delete(action.payload) && state[action.type]
+          : state[action.type].set(action.payload, action.payload),
```

Comment:
Since it seems that `key: value` in the map are always equal, can you use a `Set()` instead?

Reply:
good idea, better idea, great idea